sokutei.h

```diff
@@ -0,0 +1,188 @@
+#ifndef SOKUTEI_BENCHMARK_H
+#define SOKUTEI_BENCHMARK_H
+
+#define SOKUTEI_ITERATIONS 10
+#define SOKUTEI_MAX_COUNTER_COUNT  100
+#define SOKUTEI_COUNTER_LENGTH 30
+#define SOKUTEI_INTEGER_COUNTER_TYPE int
+#define SOKUTEI_FLOAT_COUNTER_TYPE double
+#define SOKUTEI_INTERVAL_TIMER_COUNTER_TYPE int
+
+#define MAX_SIZE_OF_TYPES   sizeof(SOKUTEI_INTEGER_COUNTER_TYPE) > sizeof(SOKUTEI_FLOAT_COUNTER_TYPE) \
+                            ? sizeof(SOKUTEI_INTEGER_COUNTER_TYPE) > sizeof(SOKUTEI_INTERVAL_TIMER_COUNTER_TYPE) \
+                                ? sizeof(SOKUTEI_INTEGER_COUNTER_TYPE) \
+                                : sizeof(SOKUTEI_INTERVAL_TIMER_COUNTER_TYPE)\
+                            : sizeof(SOKUTEI_FLOAT_COUNTER_TYPE) > sizeof(SOKUTEI_INTERVAL_TIMER_COUNTER_TYPE) \
+                                ? sizeof(SOKUTEI_FLOAT_COUNTER_TYPE) \
+                                : sizeof(SOKUTEI_INTERVAL_TIMER_COUNTER_TYPE)
+
+#define SOKUTEI_INTEGER_TYPE '1'
+#define SOKUTEI_FLOAT_TYPE '2'
+#define SOKUTEI_INTERVAL_TYPE '3'
+
+#define NOT_FOUND -1
+#define NOT_MATCHING_TYPE -2
+
+
+int sokutei_strcmp(const char *string_a, const char *string_b){
+    int i = 0;
+    while(string_a[i] == string_b[i] && string_a[i]) i++;
+    return string_a[i] - string_b[i];
+}
+#define STRCMP(string_a, string_b) sokutei_strcmp(string_a, string_b)
+
+char *sokutei_strcpy(char *string_a, const char *string_b){
+    int i;
+    for(i = 0; (string_a[i] = string_b[i]) && (i < SOKUTEI_COUNTER_LENGTH); i++);
+    string_a[SOKUTEI_COUNTER_LENGTH] = '\0';
+    return string_a;
+}
+#define STRCPY(string_a, string_b) sokutei_strcpy(string_a, string_b)
+#define string_equals(string_a, string_b) (STRCMP(string_a, string_b) == 0)
+
+
+
+
+
+
+char sokutei_counter_names[SOKUTEI_MAX_COUNTER_COUNT][SOKUTEI_COUNTER_LENGTH + 2] = {'\0'};
```

Comment:
Just by reading the code, it's not immediately clear, why `SOKUTEI_COUNTER_LENGTH + 2` is present instead of `SOKUTEI_COUNTER_LENGTH`.

I think, you should consider renaming the array so that it reflects that it not only stored counter names but also counter types. What do you think of something like `sokutei_counter_definitions`? Also, I think, adding a comment which explains how names and types are stored would be great!